### ISSUE TITLE
webMvcAsyncTaskExecutor fails to be injected when a custom ThreadPoolTaskExecutor bean exists.

### ISSUE THREAD

**Issue by thorntonrp** [codecentric/spring-boot-admin#864]
I just upgraded to spring-boot-admin-starter-server 2.0.2 and I'm getting this error:
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Method requestMappingHandlerAdapter in org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration required a bean named 'webMvcAsyncTaskExecutor' that could not be found.


Action:

Consider defining a bean named 'webMvcAsyncTaskExecutor' in your configuration.
```

To reproduce, create the following Spring Boot application:

```java
@EnableAdminServer
@SpringBootApplication
public class Main {

    public static void main(String[] args) {
        SpringApplication.run(Main.class, args);
    }

    @Bean({"threadPoolTaskExecutor", "webMvcAsyncTaskExecutor"})
    public ThreadPoolTaskExecutor threadPoolTaskExecutor() {
        return new ThreadPoolTaskExecutor();
    }
}
```
The problem is in `AdminServerWebConfiguration.ServletRestApiConfiguration`. The bean returned from `adminWebMvcConfigurer` calls into the `webMvcAsyncTaskExecutor` bean factory method which is looking for a bean of type `AsyncTaskExecutor` having the name `webMvcAsyncTaskExecutor`.  That factory method, however, has `@ConditionalOnBean`, so Spring doesn't invoke it because the bean already exists. The injection fails, however, because my bean doesn't have the name "webMvcAsyncTaskExecutor".

**Comment by thorntonrp** [comment]
If `adminWebMvcConfigurer` where defined as follows, I think it would solve the issue:

```java
@Bean
public WebMvcConfigurer adminWebMvcConfigurer(AsyncTaskExecutor asyncTaskExecutor) {
    return new WebMvcConfigurer() {
        @Override
        public void configureAsyncSupport(AsyncSupportConfigurer configurer) {
            configurer.setTaskExecutor(asyncTaskExecutor);
        }
    };
}
```

**Comment by stockmaj** [comment]
I did not define my own Executor and still ended up with this issue. The entire application is:
```
import de.codecentric.boot.admin.server.config.EnableAdminServer;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;

@EnableAdminServer
@EnableAutoConfiguration
public class SpringBootAdminApplication {
    public static void main(String[] args){
        SpringApplication.run(SpringBootAdminApplication.class, args);
    }
}
```

My dependencies are
```
dependencies {
    compile("de.codecentric:spring-boot-admin-starter-server:2.0.2")
    compile("org.springframework.boot:spring-boot-starter-web")
}
```

Got the same error as above:
```
***************************
APPLICATION FAILED TO START
***************************
Description:

Method requestMappingHandlerAdapter in org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration required a bean named 'webMvcAsyncTaskExecutor' that could not be found.

Action:

Consider defining a bean named 'webMvcAsyncTaskExecutor' in your configuration.

Process finished with exit code 1
```

**Comment by joshiste** [comment]
@stockmaj yeah, cause the fix is in 2.0.3 (not released yet)
